aten/src/ATen/core/Allocator.h

```diff
@@ -43,13 +43,19 @@ class DataPtr {
   void* release_context() {
     return ptr_.release_context();
   }
+  std::unique_ptr<void, DeleterFnPtr>&& move_context() {
+    return std::move(ptr_.move_context());
+  }
```

Comment:
This is one of those things where I'm still not really sure what idiomatic C++ design here is. I guess this should be OK for now :)